Framework/Algorithms/src/ConvertUnitsUsingDetectorTable.cpp

```diff
@@ -276,29 +120,52 @@ void ConvertUnitsUsingDetectorTable::convertViaTOF(
         "DetectorParameter TableWorkspace is not defined correctly.");
   }
 
-  // Now let's read them into some vectors.
-  auto l1Column = paramWS->getColVector<double>("l1");
-  auto l2Column = paramWS->getColVector<double>("l2");
-  auto twoThetaColumn = paramWS->getColVector<double>("twotheta");
-  auto efixedColumn = paramWS->getColVector<double>("efixed");
-  auto emodeColumn = paramWS->getColVector<int>("emode");
-  auto spectraColumn = paramWS->getColVector<int>("spectra");
-
-  EventWorkspace_sptr eventWS =
-      boost::dynamic_pointer_cast<EventWorkspace>(outputWS);
-  assert(static_cast<bool>(eventWS) == m_inputEvents); // Sanity check
+  // Now let's take a reference to the vectors.
+  const auto &l1Column = paramWS->getColVector<double>("l1");
+  const auto &l2Column = paramWS->getColVector<double>("l2");
+  const auto &twoThetaColumn = paramWS->getColVector<double>("twotheta");
+  const auto &efixedColumn = paramWS->getColVector<double>("efixed");
+  const auto &emodeColumn = paramWS->getColVector<int>("emode");
+  auto &spectraColumn = paramWS->getColVector<int>("spectra");
```

Comment:
This does not seem to be modified (lucky for us, otherwise it would edit the table since we now have a reference), so it should be `const`, I think.